src/Widgets/PartitionMenu.vala

```diff
@@ -280,22 +280,37 @@ public class Installer.PartitionMenu : Gtk.Popover {
             ? Distinst.FileSystemType.SWAP
             : get_file_system ();
 
-        set_mount (new Installer.Mount (
+        var error = set_mount (new Installer.Mount (
             partition_path,
             parent_disk,
             mount,
-            (format_partition.active ? Mount.Flags.FORMAT : 0) + (is_lvm ? Mount.Flags.LVM : 0),
+            partition_bar.end - partition_bar.start,
+            (format_partition.active ? Mount.Flags.FORMAT : 0)
+                + (is_lvm ? Mount.Flags.LVM : 0)
+                + (partition_bar.info->is_esp () ? Mount.Flags.ESP : 0),
             filesystem,
             this
         ));
 
-        var mount_icon = new Gtk.Image.from_icon_name (
-            "process-completed-symbolic",
-            Gtk.IconSize.SMALL_TOOLBAR
-        );
+        var mount_icon_name = error == null
+            ? "process-completed-symbolic"
+            : "dialog-warning-symbolic";
+
+        var icon_theme = Gtk.IconTheme.get_default ();
+        var pixbuf = icon_theme.load_icon (mount_icon_name, 16, 0);
+
+        if (error != null) {
```

Comment:
It looks like we can just do this in the Gtk CSS theme which is probably the best and most flexible route. i.e. for the storage bars, we can set `-gtk-icon-palette: success #fff, warning #fff, error #fff;` to make sure it contrasts, or we can even set it to different colors for each filesystem type.